modules/ml/src/main/java/org/apache/ignite/ml/inference/reader/DirectoryInfModelReader.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.inference.reader;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import org.apache.ignite.ml.inference.util.DirectorySerializer;
+
+/**
+ * Model reader that reads directory and serializes it using {@link DirectorySerializer}.
+ */
+public class DirectoryInfModelReader implements InfModelReader {
```

Comment:
Seems like we could avoid separate classm and check path.isDirectory in common approach

Reply:
Good point, done (see `FileSystemInfModelReader`).